contracts/tgrade-community-pool/src/contract.rs

```diff
@@ -104,11 +102,15 @@ pub fn execute_execute(
     use Proposal::*;
 
     // anyone can trigger this if the vote passed
-    let prop = mark_executed::<Proposal>(deps, env, proposal_id)?;
+    let prop = mark_executed::<Proposal>(deps.storage, env, proposal_id)?;
 
     // dispatch all proposed messages
     let resp = match prop.proposal {
         SendProposal { to_addr, amount } => execute_send_proposal(to_addr, amount)?,
+        Text {} => {
+            execute_text(deps, proposal_id, &prop)?;
```

Comment:
Returning a Response seemed like it would require more refactoring in `execute_execute` of `tgrade-validator-voting`. Also feels like it's generally the `execute_execute` function that chooses what events/attributes are emitted for a given execution. Felt more natural at a glance, but I didn't think too deeply about it.

Reply:
Fair enough. When I read through the other usages, I understood the design better